simulator/src/delegate.rs

```diff
@@ -25,24 +29,23 @@ impl Delegate {
     #[tracing::instrument(skip_all)]
     pub fn emit(&mut self, asset_id: Hash, amount: u64) -> Result<Note, Error> {
         let mut note = Note {
-            delegate: self.keypair.public_key,
+            delegate: self.keypair.public(),
             asset_id,
             nonce: Hash::random(&mut self.rng),
             amount,
             signature: Signature::default(),
         };
 
         let blind = crypto::blind_note(&mut self.rng, &note);
-        let signed = crypto::sign_blinded(&self.keypair.secret_key, &blind.point);
-        note.signature =
-            crypto::unblind_signature(&signed, &blind.factor, &self.keypair.public_key)?;
+        let signed = crypto::sign_blinded::<SchnorrPair>(self.keypair.secret(), &blind.point);
+        note.signature = crypto::unblind_signature(&signed, &blind.factor, &self.keypair.public())?;
 
         Ok(note)
     }
 
     #[inline(always)]
     #[tracing::instrument(skip_all)]
     pub fn recv_transaction_v0(&mut self, tx: &Transaction) -> Result<V0Response, Error> {
-        transaction_v0(tx, self.keypair, &mut self.db)
+        transaction_v0(tx, self.keypair.clone(), &mut self.db)
```

Comment:
You can pass this as a reference on the method, no?

Reply:
You're right, I've amended it.